security/security.c

```diff
@@ -86,6 +100,67 @@ static __initconst const char * const builtin_lsm_order = CONFIG_LSM;
 static __initdata struct lsm_info **ordered_lsms;
 static __initdata struct lsm_info *exclusive;
 
+/*
+ * Necessary information about a static
+ * slot to call __static_call_update
+ */
+struct security_static_slot {
+	/* static call key as defined by STATIC_CALL_KEY */
+	struct static_call_key *key;
+	/* static call tramp as defined by STATIC_CALL_TRAMP */
+	void *tramp;
+};
+
+/* Table of the static calls for each LSM hook */
+struct security_list_static_slots {
+	#define LSM_HOOK(RET, DEFAULT, NAME, ...) \
+		struct security_static_slot NAME[SECURITY_STATIC_SLOT_COUNT];
+	#include <linux/lsm_hook_defs.h>
+	#undef LSM_HOOK
+} __randomize_layout;
+
+/*
+ * Index of the first used static call for each LSM hook
+ * in the corresponding security_list_static_slots table.
+ * All calls with greater indices are used, INT_MAX if no call is used.
+ */
+struct security_first_static_slot_idx {
```

Comment:
This is better named as lsm_hook_base_slot or even just just base_slot_info (you don't need security_* or lsm_hook_ because this struct is local to this file.